api/model/api.go

```diff
@@ -60,7 +60,7 @@ type (
 type HttpMethod string
 
 // GraphQLExecutionMode is the mode to define how an api behaves.
-// +kubebuilder:validation:Enum=proxyOnly;executionEngine;supergraph;subgraph
+// +kubebuilder:validation:Enum="";proxyOnly;executionEngine;supergraph;subgraph
```

Comment:
Could you explain why we are allowing empty value here for GraphQLExecutionMode?

Reply:
similar reasons as `APIProtocol` field. 
Once APIs are exported, all fields come with their nil values if they are not explicitly declared. So, although users do not use any GraphQL features, they cannot apply their manifests due to the empty value of `GraphQLExecutionMode` which is mandatory.